src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -144,7 +160,89 @@ public void Write(JsonTextWriter writer)
             return (Template.FromJson<Template>(content), content.FromJson<JToken>());
         }
 
-        private void EmitParametersIfPresent(JsonTextWriter jsonWriter, ExpressionEmitter emitter)
+        private void ProcessRawSourceMap(JToken rawTemplate)
```

Comment:
Please could you make this into a static helper method and move into a different class (`SourceMapHelper`?)